### ISSUE TITLE
docs(architecture): codify neuromorphic/SNN scope contract

### ISSUE THREAD

**Issue by rmems** [Limen-Neural/limbic-critic#58]
## Summary

Codify the long-term architectural contract that keeps `limbic-critic` explicitly within the **neuromorphic / spiking-neural-network** domain rather than drifting into a generic reinforcement-learning or ML utility crate.

This is a scope-governance issue, not a request to expand functionality before the first crates.io release.

## Core contract

`limbic-critic` exists to transform environment feedback into bounded, biologically inspired modulatory signals intended for **spiking/plasticity systems**.

Canonical flow:

```text
Environment
    ↓
SimpleCritic / TDCritic / future critic
    ↓
ModulatorVector
(dopamine / serotonin / acetylcholine / norepinephrine)
    ↓
plasticity / neuromodulatory bridge
    ↓
SNN / neuromorphic runtime
```

### Owns

- Reward shaping / objective-delta shaping for SNN-oriented systems
- Mapping environment signals into neuromodulator-like bounded scalars
- `Environment` as the observation boundary
- `ModulatorVector` as the crate-local output contract
- Critic algorithms whose output is a neuromodulatory control signal

### Does not own

- Generic actor–critic loops
- Learned policy/value networks
- Policy gradients / PPO / SAC / DQN or other general RL algorithms
- Full training loops
- ANN model definitions
- Gym-style environment orchestration
- Domain-specific controllers or reward implementations
- Synapse / neuron / STDP implementations owned by sibling neuromorphic crates

## Future-feature rule

A new critic or public abstraction belongs in this crate only if its primary output or role can be expressed as a **neuromodulatory / plasticity control signal for spiking systems**.

Features that instead produce actions, policies, value estimates, generic episode management, or ANN-specific training behavior should live elsewhere.

## Tasks

- [ ] Add an explicit `Neuromorphic / SNN Scope Contract` section to `docs/BOUNDARY_MATRIX.md`
- [ ] Add a condensed scope statement to the README
- [ ] Document the canonical `Environment → Critic → ModulatorVector → plasticity/SNN` flow
- [ ] Document the future-feature admission rule
- [ ] Clarify that semantic interoperability with `plasticity-lab` / `neuromod` is intentional while Cargo-level sibling dependencies remain forbidden
- [ ] Ensure examples reinforce SNN / neuromodulatory usage without making the crate depend on a sibling repo

## Acceptance criteria

- [ ] A cold reader can explain why `limbic-critic` belongs in Limen-Neural
- [ ] The docs positively define the crate as neuromorphic/SNN-oriented, not merely “not RL”
- [ ] Generic RL-framework scope is explicitly rejected
- [ ] Future feature proposals have a documented architectural test for whether they belong here
- [ ] No sibling Cargo dependencies are added
- [ ] No new critic algorithms are required by this issue

## Relationships

- Related: #40 — crates.io first-publish readiness
- Related: #47 — `0.2.0` release
- Related: `docs/BOUNDARY_MATRIX.md`
- Intended long-term home: `Limen-Neural/limbic-critic`

Cited by: ChatGPT / GPT-5.6 Sol architecture audit (2026-09-13)
